### ISSUE TITLE
Provide rpm repos via openSUSE Build Service.

### ISSUE THREAD

**Issue by sudopluto** [atom/atom#6509]
As of right now, there is no way to update atom on linux besides manually downloading and installing a .deb or a .rpm file. While there is support for adding a .deb repo via launchpad, there has been no support for .rpm distributions. Hosting a repository on the openSUSE build service will provide both Fedora and openSUSE users with an option to update atom via package manager.

**Comment by SudoPluto** [comment]
https://build.opensuse.org/
https://en.opensuse.org/openSUSE:Build_Service

**Comment by SudoPluto** [comment]
http://www.makehuman.org/doc/node/packaging_rpms_for_susefedora_using_open_build_service_obs.html

**Comment by pyrokinetiq** [comment]
:+1:

**Comment by Yajo** [comment]
I'm subscribed to https://github.com/atom/atom/tags.atom to know when I can update and do it manually. It's boring. Fedora here.

**Comment by sudopluto** [comment]
So since nobody wants to touch this I am going to try this by myself. I wanted to avoid this since I am kinda new to Linux and I have never compiled something before. If you want to help pls let me know.

**Comment by amayer5125** [comment]
@sudopluto Did you ever get this working on open suse build service?

**Comment by sudopluto** [comment]
@amayer5125 I haven't tried it yet. I read from somewhere that atom needs an internet connection to be built. Right now the only build service on Linux that has an internet connection is Copr. There is also the fact I had just switched to linux in April, and I still have no clue how to package.

TL;RD : No.

**Comment by amayer5125** [comment]
I found a repo link in another issue thread. They just host the prebuilt ones from atom/atom/releases which if fine by me.

**Comment by mnquintana** [comment]
Do apps hosted on openSUSE (or any other software repository for that matter) need to be built from source? Or is it fine for them to just use the prebuilt binaries?

**Comment by amayer5125** [comment]
@mnquintana I guess it depends on the hosting.
If you are looking for a temporary repo you can use [this one](www.a9f.eu). I hashed the most recent version hosted in the rpm vs the one on atom/atom/releases and the hases match up so I guess his server is just grabbing the rpm/debs and hosting them in a repo, which works perfect. I havent used it for an update yet but I installed from it using dnf (yum replacemnt) on fedora 22.

**Comment by Yajo** [comment]
You can build RPM from binaries using nosrc RPMs, but being an open source project that's not very pleasing, and you could automate many things.

RPM normal packages can have several sources and patches included, so there should be no need to have Internet connection to build. If it is, then IMHO it's a bug in Atom.

**Comment by jubalh** [comment]
It seems someone has a spec file for Fedora here: https://raw.githubusercontent.com/LukeCarrier/rpm-specs/master/SPECS/atom.spec
And here you can find the Requires/BuildRequires for openSUSE: https://github.com/atom/atom/blob/master/docs/build-instructions/linux.md#opensuse

If I would know npm better I would package it myself, but I don't know enough about it.

**Comment by fusion809** [comment]
@sudopluto this is still true, that no means exists to automatically update Atom on RPM platforms. I have created a spec file for Atom 1.5.2 that works without issue on platforms with a valid internet connection that is:

```
Name:           atom
Version:        1.5.2
Release:        0
Summary:        A hackable text editor for the 21st century
License:        MIT
Group:          Productivity/Publishing/Other
Url:            https://atom.io/
Source0:        v%{version}.tar.gz
BuildRequires:  git-core
BuildRequires:  hicolor-icon-theme
BuildRequires:  npm
BuildRequires:  nodejs-packaging
BuildRequires:  libgnome-keyring-devel
BuildRequires:  python-setuptools
BuildRequires:  update-desktop-files
# MANUAL BEGIN
Requires:       nodejs
Requires:       python-http-parser
# MANUAL END
BuildRoot:      %{_tmppath}/%{name}-%{version}-build

%description
Atom is a text editor that's modern, approachable, yet hackable to the core
- a tool you can customize to do anything but also use productively without
ever touching a config file.

%prep
%setup -q
sed -i -e "s/<%= appName %>/Atom/g" \
       -e "s/<%= description %>/Atom is a text editor that's modern, approachable, yet hackable to the core - a tool you can customize to do anything but also use productively without ever touching a config file./g" \
       -e "s|<%= installDir %>/share/<%= appFileName %>/atom %U|/usr/bin/atom %U|g" \
       -e "s/<%= iconPath %>/atom/g" \
       resources/linux/atom.desktop.in
%build
# Hardened package
export CFLAGS="%{optflags} -fPIC -pie"
export CXXFLAGS="%{optflags} -fPIC -pie"
until ./script/build 2>&1; do :; done

%install
script/grunt install --install-dir "%{buildroot}%{_prefix}"
# copy over icons in sizes that most desktop environments like
for i in 1024 512 256 128 64 48 32 24 16; do
    install -Dm 0644 /tmp/atom-build/icons/${i}.png \
      %{buildroot}%{_datadir}/icons/hicolor/${i}x${i}/apps/%{name}.png
done
%suse_update_desktop_file %{name}

%post
%desktop_database_post
%icon_theme_cache_post

%postun
%desktop_database_postun
%icon_theme_cache_postun

%files
%defattr(-,root,root,-)
%doc CONTRIBUTING.md README.md docs/
%{license} LICENSE.md
%{_bindir}/atom
%{_bindir}/apm
%dir %{_datadir}/atom
%{_datadir}/atom/*
%{_datadir}/atom
%{_datadir}/applications/atom.desktop
%{_datadir}/icons/hicolor/*/apps/%{name}.png
%exclude %{_datadir}/%{name}/libgcrypt.so.*
%exclude %{_datadir}/%{name}/libnotify.so.*

%changelog
```

if it helps. I asked a similar question to this one at forums.opensuse.org too https://forums.opensuse.org/showthread.php/513540-How-to-create-a-spec-file-for-Atom-that-is-suitable-for-the-OBS?p=2753770.

**Comment by TJProgrammer** [comment]
I have find this simple, I guess, solution.
The Yarn project has the following instructions for install in Fedora, for example:
> On CentOS, Fedora and RHEL, you can install Yarn via our RPM package repository.
> sudo wget https://dl.yarnpkg.com/rpm/yarn.repo -O /etc/yum.repos.d/yarn.repo
> Then you can simply:
> sudo yum install yarn

I got the yarn.repo and it is like this:
```
[yarn]
name=Yarn Repository
baseurl=https://dl.yarnpkg.com/rpm/
enabled=1
gpgcheck=1
gpgkey=https://dl.yarnpkg.com/rpm/pubkey.gpg
```
Let me know if I helped ...

**Comment by fusion809** [comment]
@TJProgrammer how is that relevant to this question? We're not asking how to install Yarn on Fedora, we're asking the Atom development team to provide an OBS repository for Atom.

**Comment by TJProgrammer** [comment]
I am sorry for the mistake.
I take the yarn to an example to atom.
In other words, atom team can do the update system like yarn team do.
But, again, sorry for the last comment.

**Comment by 50Wliu** [comment]
I believe this can be closed now that there's an official package repository at https://packagecloud.io/AtomEditor/atom.  Please tell me if that's not the case!

**Comment by TJProgrammer** [comment]
OK. Thanks @50Wliu for the info, but ...
1. In the latest Fedora versions the `dnf` is the new standard (over `yum`).
2. I installed the repos, but there is no packages in there (when I run `dnf repolist` the number is 0 in `AtomEditor_atom` and `AtomEditor_atom-source` repos).
3. And I can't update/install to the latest AtomEditor version:
```
sudo yum install atom-1.23.3-0.1.x86_64
Last Metadata Expiration Selection: 0:01:22 Mon Mon 15 Jan 2018 00:37:00 WET.
No matches for the argument: atom-1.23.3-0.1.x86_64
Error: No match found matching
```
If I am doing something wrong, please tell me.
Thanks in advance!

**Comment by kostyrev** [comment]
```
yum update
yum --showduplicates list atom
```

**Comment by TJProgrammer** [comment]
The result is:
```
[root@toshiba ~]# dnf --showduplicates list atom
Pacotes Instalados
atom.x86_64                       1.23.1-0.1                       @@commandline
Pacotes Disponíveis
atom.x86_64                       1.23.1-0.1                       @@commandline
```
P.S.: My system is in portuguese.

**Comment by TJProgrammer** [comment]
The answer is in [Installing Atom](http://flight-manual.atom.io/getting-started/sections/installing-atom/#platform-linux) instructions.
